lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+    end
+
+    # returns the path of the docker service
+    # only works for Debian 8+, Ubuntu 16.04+ and CentOS 7+ where systemd is installed
+    def path
+      # TODO: use exception, once https://github.com/chef/inspec/issues/1205 is fixed
+      return nil unless is_systemd_system
+
+      cmd = inspec.command('systemctl show -p FragmentPath docker.service')
+      return nil if cmd.exit_status.to_i.nonzero?
+
+      # parse data
+      params = parse_systemd_values(cmd.stdout.chomp)
+
+      # return the value
+      params['FragmentPath']
+    end
+
+    # returns the docker socket
+    # only works for Debian 8+, Ubuntu 16.04+ and CentOS 7+ where systemd is installed
+    def socket
+      # TODO: use exception, once https://github.com/chef/inspec/issues/1205 is fixed
+      return nil unless is_systemd_system
+
+      cmd = inspec.command('systemctl show -p FragmentPath docker.socket')
+      return nil if cmd.exit_status.to_i.nonzero?
+
+      # parse data
+      params = parse_systemd_values(cmd.stdout.chomp)
+
+      # return the value
+      params['FragmentPath']
+    end
+
+    def to_s
+      'Docker Host'
+    end
+
+    # returns the json output of `docker inspect`
+    def inspect(id)
+      return @inspect if defined?(@inspect)
+      data = JSON.parse(inspec.command("docker inspect #{id}").stdout)
+      # we'll return the first value only
+      data = data[0] if data.is_a?(Array)
+      @inspect = Hashie::Mash.new(data)
+    end
+
+    # returns the json object of the info
+    def version
+      return @version if defined?(@version)
+      data = JSON.parse(inspec.command('docker version --format \'{{ json . }}\'').stdout)
+      @version = Hashie::Mash.new(data)
+    end
+
+    private
+
+    def parse_containers
+      raw_containers = inspec.command('docker ps -a --no-trunc --format \'{{ json . }}\'').stdout
+      ps = []
+      # since docker is not outputting valid json, we need to parse each row
+      raw_containers.each_line { |entry|
+        j = JSON.parse(entry)
+        # convert all keys to lower_case to work well with ruby and filter table
+        j = j.map {|k, v|
+          [k.downcase, v]
+        }.to_h
+        ps.push(j)
+      }
+      ps
+    end
+
+    def parse_images
+      # docker does not support the `json .` function here, therefore we need to emulate that behavior.
+      raw_images = inspec.command('docker images -a --no-trunc --format \'{ "id": {{json .ID}}, "repository": {{json .Repository}}, "tag": {{json .Tag}}, "size": {{json .Size}}, "digest": {{json .Digest}}, "createdat": {{json .CreatedAt}}, "createdsize": {{json .CreatedSince}} }\'').stdout
+      c_images = []
+      raw_images.each_line { |entry|
+        c_images.push(JSON.parse(entry))
+      }
+      c_images
+    end
+
+    # checks if the system is Debian 8+, Ubuntu 16.04+ and RHEL 7+
+    def is_systemd_system
+      return true if inspec.os.redhat? && inspec.os.release.to_i >= 7
```

Comment:
difficult, since init system is not necessarily tight to a specific operating system. Chef is the best example and uses runit. I know we need to extend this in future. Maybe we kick this out of the resource, let me think about it